src/irmin-pack/unix/ext.ml

```diff
@@ -351,6 +354,30 @@ module Maker (Config : Conf.S) = struct
                       Pack_key.v_direct ~offset ~length ~hash:entry.hash
                     in
                     Some key)
+
+          let create_one_commit_store t commit_key path =
+            let () = Io.mkdir path |> Errs.raise_if_error in
+            let* _launched =
+              start_exn ~use_auto_finalisation:false ~path_new_files:path t
+                commit_key
+            in
+            let generation = File_manager.generation t.fm + 1 in
+            let* () =
+              match t.running_gc with
+              | None -> assert false
```

Comment:
I added a ` Cannot_run_during_gc` in case the gc is not launched, but I kept the assert false in `t.running = None` - if there is a gc running then we raise the error above, otherwise the gc was launched, and so I think the assert false here is appropriate. do you agree?